### DISCUSSION TITLE
openapi_core.unmarshalling.schemas.exceptions.InvalidSchemaValue

### DISCUSSION THREAD

**Discussion by itsMGA** [p1c2u/openapi-core#532] [Ideas]
Atm, it seems for any schema irregularities, lets say in a massive json, where a integer field is string on response, it only raises this error without the specific field that is causing the problem, and figuring out which field is problematic is tedious, could we add a better schema validations errors messages?

I m using the following approach to validate the response vs the yaml schema:
```
        request = RequestsOpenAPIRequest(response.request)
        response = RequestsOpenAPIResponse(response)
        validate_response(response=response, spec=self.get_partner_spec(), request=request)

```

**Comment by p1c2u** [comment]
Hi @itsMGA 

thanks for reporting the issue. Can you send more details? What version do you use? What exact error you get? Can you paste working  example? That would help debugging your problem.

**Comment by gsauthof** [comment]
I think the issue is that the default InvalidRequestBody exception message string doesn't include _some_ schema validation context.

Arguably, one has to go to some lengths to get a hint about the root cause:

```
oa = openapi_core.OpenAPI.from_file_path('some.yaml')
r = requests.Request('POST', url, json=d, headers={ 'Authorization': f'Bearer {token}' })
ro = openapi_core.contrib.requests.RequestsOpenAPIRequest(r)
try:
    oa.validate_request(ro)
except openapi_core.validation.request.exceptions.InvalidRequestBody as e:
    print(e.__cause__.schema_errors[0].context)
```

For example, in one case, without the above approach, I get something like this:

`InvalidSchemaValue: Value { .... extremely long request json body ...  }  not valid for schema of type any: (<ValidationError: "{ ... same extremely long request json body ...} is not valid under any of the given schemas">,)`

While the context includes a _much_ more helpful error message:

`[<ValidationError: "{' ... request json body ... } reference '#/components/schemas/foobar' could not be resolved">,
 <ValidationError: "'fieldxyz' is a required property">`

Curiously, `OpenAPI.check_spec()` doesn't report any errors.
Apparently it doesn't check the references used in schemas ...

---

FWIW, the above reference error was caused by a nonsensically placed `discriminator` object (i.e. adjacent to an allOf construct), where on the one hand openapi-core marks discriminators as unsupported, and discriminator objects aren't suppose to influence the validation outcome - but on the other hand openapi-schema-validator verifies discriminator references ...:

https://github.com/python-openapi/openapi-schema-validator/blob/bf12c9bf5130fac2768d13a9010ede8fa7ec2aa4/openapi_schema_validator/_keywords.py#L40-L61

And this fails when the discriminator property points to itself, in an allOf-adjacent context.
